Automattic/wp-calypso
# PlansGrid: Use selected plan's billing period to initialize billingPeriod state

#### Changes proposed in this Pull Request

* Use `selectedPlanBillingPeriod` to set initial value for `billingPeriod` state in PlanGrid as described in https://github.com/Automattic/wp-calypso/issues/49339#issuecomment-768919166

#### Testing instructions

* Go to /new/design and open Plans Modal (top-right button) => yearly billing period is default
* Toggle to monthly and go back without making a selection. Open the modal again =>  yearly billing period is default
* This time select a monthly plan. Open the modal again =>  monthly billing period is default
* The same should apply for post-Gutenboarding launch (Select a Plan step) and Focused Launch (Plan Detail view).

Related to #49339
